libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -267,6 +267,9 @@ - (void)replayRequestIfRequired:(NSData *)data response:(NSURLResponse *)respons
             NSError *parsingError;
             NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&parsingError];
             if (parsingError) {
+                if (data == nil || [data isEqualToData:[[NSData alloc] init]]) {
```

Comment:
Can we check the `data.length` property?  Shouldn't need to compare to an empty object.